File: ios/Donna/DonnaShareInbox.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(DonnaShareInbox, NSObject)

RCT_EXTERN_METHOD(takePending:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/Donna/DonnaShareInbox.swift
import Foundation
import React

@objc(DonnaShareInbox)
final class DonnaShareInbox: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {
    false
  }

  @objc func takePending(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {
    resolve(DonnaShareStore.takePending())
  }
}
